tests/e2e/backup_restore_suite_test.go

```diff
@@ -50,20 +40,63 @@ var _ = Describe("AWS backup restore tests", func() {
 
 	})
 
-	type VerificationFunction func(client.Client, string) error
-
 	type BackupRestoreCase struct {
 		ApplicationTemplate  string
 		ApplicationNamespace string
 		Name                 string
+		BackupRestoreType    BackupRestoreType
 		PreBackupVerify      VerificationFunction
 		PostRestoreVerify    VerificationFunction
 		MaxK8SVersion        *k8sVersion
 		MinK8SVersion        *k8sVersion
 	}
 
+	parksAppReady := VerificationFunction(func(ocClient client.Client, namespace string) error {
+		Eventually(isDCReady(ocClient, "parks-app", "restify"), timeoutMultiplier*time.Minute*10, time.Second*10).Should(BeTrue())
+		return nil
+	})
+	mssqlReady := VerificationFunction(func(ocClient client.Client, namespace string) error {
+		// This test confirms that SCC restore logic in our plugin is working
+		Eventually(isDCReady(ocClient, "mssql-persistent", "mssql-deployment"), timeoutMultiplier*time.Minute*10, time.Second*10).Should(BeTrue())
+		Eventually(isDeploymentReady(ocClient, "mssql-persistent", "mssql-app-deployment"), timeoutMultiplier*time.Minute*10, time.Second*10).Should(BeTrue())
+		exists, err := doesSCCExist(ocClient, "mssql-persistent-scc")
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return errors.New("did not find MSSQL scc")
+		}
+		return nil
+	})
+
 	DescribeTable("backup and restore applications",
 		func(brCase BackupRestoreCase, expectedErr error) {
+
+			err := vel.Build(brCase.BackupRestoreType)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = vel.CreateOrUpdate(&vel.CustomResource.Spec)
+			Expect(err).NotTo(HaveOccurred())
+
+			log.Printf("Waiting for velero pod to be running")
+			Eventually(isVeleroPodRunning(namespace), timeoutMultiplier*time.Minute*3, time.Second*5).Should(BeTrue())
+
+			if brCase.BackupRestoreType == restic {
+				if vel.CustomResource.Spec.EnableRestic == nil || *vel.CustomResource.Spec.EnableRestic {
+					log.Printf("Waiting for restic pods to be running")
+					Eventually(areResticPodsRunning(namespace), timeoutMultiplier*time.Minute*3, time.Second*5).Should(BeTrue())
+				}
+			}
+			if brCase.BackupRestoreType == csi {
+				log.Printf("Creating VolumeSnapshot for CSI backuprestore of %s", brCase.Name)
+				err = installApplication(vel.Client, "./sample-applications/mssql-persistent/volumeSnapshotClass.yaml")
```

Comment:
Just a NIT: I would move this path to `./sample-applications/gp2-csi/volumeSnapshotClass.yaml` just to explictly mention this volumesnapshotclass is for gp2-csi